File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        
        Map<Integer, Integer> hashmap = new HashMap<>();
        int result[] = new int[k];


        for (int i : nums){
            hashmap.put(i, hashmap.getOrDefault(i, 0) + 1);
        }

        PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a,b) -> Integer.compare(hashmap.get(b), hashmap.get(a)));

        for(int key : hashmap.keySet()){
            maxHeap.add(key);
        }
        
        for(int i = 0; i < k; i++){
            result[i] = maxHeap.poll();
        }

        return result;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        hashmap = collections.defaultdict(int)
        result = []

        for i in nums:
            hashmap[i] += 1

        for num, freq in hashmap.items():
            result.append([freq, num])

        result.sort()

        return [num for freq, num in result[-k:]]